dask/threaded.py

```diff
@@ -45,16 +52,28 @@ def get(dsk, result, cache=None, num_workers=None, **kwargs):
     (4, 2)
     """
     pool = _globals['pool']
+    thread = current_thread()
 
     if pool is None:
-        if num_workers:
-            pool = ThreadPool(num_workers)
-        else:
+        if num_workers is None and thread is main_thread:
             pool = default_pool
+        elif thread in pools and num_workers in pools[thread]:
+            pool = pools[thread][num_workers]
+        else:
+            pool = ThreadPool(num_workers)
+            pools[thread][num_workers] = pool
```

Comment:
Don't you need a lock here? I'm imagining a race condition where multiple threads try to create thread pools at the same time, e.g., there wasn't an existing pool when the check on line 61 was run, but there is when you overwrite the dictionary key.